Applications/Utils/FileConverter/generateMatPropsFromMatID.cpp

```diff
@@ -52,47 +52,58 @@ int main (int argc, char* argv[])
     cmd.parse( argc, argv );
 
     // read mesh
-    std::unique_ptr<MeshLib::Mesh> mesh(MeshLib::IO::readMeshFromFile(mesh_arg.getValue()));
-    if (!mesh) {
-        INFO("Could not read mesh from file \"%s\".", mesh_arg.getValue().c_str());
-        return EXIT_FAILURE;
-    }
-    if (!mesh->getProperties().existsPropertyVector<int>("MaterialIDs"))
-    {
-        ERR("Mesh contains no int-property vector named \"MaterialIds\".");
-        return EXIT_FAILURE;
-    }
-    auto materialIds = mesh->getProperties().getPropertyVector<int>("MaterialIDs");
+    std::unique_ptr<MeshLib::Mesh> mesh(
+        MeshLib::IO::readMeshFromFile(mesh_arg.getValue()));
 
-    std::size_t const n_properties(materialIds->size());
-    if (n_properties != mesh->getNumberOfElements()) {
-        ERR("Size mismatch: number of elements (%u) != number of material "
-            "properties (%u).", mesh->getNumberOfElements(), n_properties);
+    if (!mesh)
+    {
+        INFO("Could not read mesh from file \"%s\".", mesh_arg.getValue().c_str());
         return EXIT_FAILURE;
     }
-    std::string const name = BaseLib::extractBaseNameWithoutExtension(mesh_arg.getValue());
-    // create file
-    std::string const new_matname(name + "_prop");
-    std::ofstream out_prop( new_matname.c_str(), std::ios::out );
-    if (out_prop.is_open())
+    try
     {
-        for (std::size_t i=0; i<n_properties; ++i)
-            out_prop << i << "\t" << (*materialIds)[i] << "\n";
-        out_prop.close();
+        auto materialIds = mesh->getProperties().getPropertyVector<int>(
+            "MaterialIDs", MeshLib::MeshItemType::Cell, 1);
+
+        std::size_t const n_properties(materialIds->size());
+        if (n_properties != mesh->getNumberOfElements())
```

Comment:
This seems to be already checked by getPropertyVector because of the Cell requirement...

Reply:
Will change it to assert.